src/libnet_build_802.2.c

```diff
@@ -114,7 +114,7 @@ libnet_t *l, libnet_ptag_t ptag)
     _802_2_hdr.snap_type = htons(type);
 
     n = libnet_pblock_append(l, p, (uint8_t *)&_802_2_hdr, LIBNET_802_2SNAP_H);
-    if (n == -1)
+    if (n == UINT32_MAX)
```

Comment:
Instead of changing `-1` to compare with `UINT32_MAX` everywhere, I'd prefer introducing a new local variable `int rc;`, or something, and using that.  Here the code would be:

```
   int rc;
   ...
   rc = libnet_pblock_append(...);
   if (rc == -1)
   ...
```

Reply:
I'll continue using C89 (not mixing code and variable declarations), but is that "official" that libnet is C89?

I'll push a new version soon with these updates.